hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -466,61 +466,17 @@ public static HoodieTableConfig loadFromHoodieProps(HoodieStorage storage, Stora
     return new HoodieTableConfig(storage, metaPath);
   }
 
-  private HoodieTableConfig(HoodieStorage storage, StoragePath metaPath) {
-    this(storage, metaPath, null, null, null, false);
-  }
-
-  public HoodieTableConfig(HoodieStorage storage, StoragePath metaPath, RecordMergeMode recordMergeMode, String payloadClassName,
-                           String recordMergeStrategyId) {
-    this(storage, metaPath, recordMergeMode, payloadClassName, recordMergeStrategyId, true);
-  }
-
-  public HoodieTableConfig(HoodieStorage storage, StoragePath metaPath, RecordMergeMode recordMergeMode, String payloadClassName,
-                           String recordMergeStrategyId, boolean autoUpdate) {
+  public HoodieTableConfig(HoodieStorage storage, StoragePath metaPath) {
     super();
     StoragePath propertyPath = new StoragePath(metaPath, HOODIE_PROPERTIES_FILE);
     LOG.info("Loading table properties from " + propertyPath);
     try {
       this.props = fetchConfigs(storage, metaPath, HOODIE_PROPERTIES_FILE, HOODIE_PROPERTIES_FILE_BACKUP, MAX_READ_RETRIES, READ_RETRY_DELAY_MSEC);
-      if (autoUpdate) {
-        autoUpdateHoodieProperties(storage, metaPath, recordMergeMode, payloadClassName, recordMergeStrategyId);
-      }
     } catch (IOException e) {
       throw new HoodieIOException("Could not load properties from " + propertyPath, e);
     }
   }
 
-  private void autoUpdateHoodieProperties(HoodieStorage storage, StoragePath metaPath,
-                                          RecordMergeMode recordMergeMode, String payloadClassName,
-                                          String recordMergeStrategyId) {
-    StoragePath propertyPath = new StoragePath(metaPath, HOODIE_PROPERTIES_FILE);
-    try {
-      boolean needStore = false;
-      if (contains(PAYLOAD_CLASS_NAME) && payloadClassName != null
```

Comment:
So basically the logic disallow changes of any merge configs (0.x releases used to allow payload class change), which make sense in terms of data consistency.